components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/confirmation/ResendConfirmationManager.java

```diff
@@ -105,6 +114,232 @@ public NotificationResponseBean resendConfirmationCode(User user, String code, S
         return validateAndResendNotification(user, code, recoveryScenario, recoveryStep, notificationType, properties);
     }
 
+    /**
+     * Resend confirmation information for the user bound to the resend code. The user will be notified via a channel
+     * recovered from the recovery data of the user.
+     *
+     * @param tenantDomain         Tenant domain
+     * @param resendCode           Previously issued confirmation code
+     * @param recoveryScenario     Name of the recovery scenario
+     *                             {@link org.wso2.carbon.identity.recovery.RecoveryScenarios}
+     * @param recoveryStep         Name of the recovery step {@link org.wso2.carbon.identity.recovery.RecoverySteps}
+     * @param notificationScenario Notification template name related to the recovery scenario (Eg: org.wso2.carbon
+     *                             .identity.recovery.IdentityRecoveryConstants
+     *                             .NOTIFICATION_TYPE_RESEND_PASSWORD_RESET
+     * @param properties           Meta properties
+     * @return ResendConfirmationDTO {@link ResendConfirmationDTO} bean resend operation information
+     * @throws IdentityRecoveryException Error while sending confirmation info
+     */
+    public ResendConfirmationDTO resendConfirmation(String tenantDomain, String resendCode, String recoveryScenario,
+                                                    String recoveryStep, String notificationScenario,
+                                                    Property[] properties) throws IdentityRecoveryException {
+
+        RecoverySteps step = RecoverySteps.getRecoveryStep(recoveryStep);
+        RecoveryScenarios scenario = RecoveryScenarios.getRecoveryScenario(recoveryScenario);
+        UserAccountRecoveryManager userAccountRecoveryManager = UserAccountRecoveryManager.getInstance();
+        // Get Recovery data.
+        UserRecoveryData userRecoveryData = userAccountRecoveryManager
+                .getUserRecoveryData(resendCode, RecoverySteps.RESEND_CONFIRMATION_CODE);
+        User user = userRecoveryData.getUser();
+        if (!tenantDomain.equals(user.getTenantDomain())) {
```

Comment:
This method seems too long. Shall we move the validation logic to a new method? e.g. validateInputs()